common/client_test.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"github.com/jarcoal/httpmock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	pb "github.com/rode/rode/proto/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"net"
+	"net/http"
+	"strings"
+)
+
+var _ = Describe("client", func() {
+	var (
+		actualRodeClient pb.RodeClient
+		actualError      error
+
+		expectedConfig *ClientConfig
+
+		fakeListener *bufconn.Listener
+		fakeServer   *grpc.Server
+
+		actualAuthenticationHeader string
+	)
+
+	BeforeEach(func() {
+		httpmock.Activate()
+		httpmock.ActivateNonDefault(insecureOauthHttpClient)
+
+		fakeAuthnFunc := func(ctx context.Context) (context.Context, error) {
+			actualAuthenticationHeader = metautils.ExtractIncoming(ctx).Get("authorization")
```

Comment:
Even though it's what's used for authentication, it feels weird to assign the authorization header to a variable named `actualAuthenticationHeader`